modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/FingerprintProcessor.java

```diff
@@ -253,7 +254,11 @@ public FingerprintProcessor create(
             }
             ThreadLocal<Hasher> threadLocalHasher = ThreadLocal.withInitial(() -> {
                 try {
-                    return MessageDigestHasher.getInstance(method);
+                    if (method.equalsIgnoreCase(MurmurHasher.METHOD)) {
+                        return MurmurHasher.getInstance(method);
+                    } else {
+                        return MessageDigestHasher.getInstance(method);
+                    }
```

Comment:
I think I would prefer this logic to live inside of `MessageDigestHasher.getInstance`, so that callers could retrieve the Murmur hasher without having to do this if check (it looks just like another built-in `MessageDigestHasher`).

What do you think about moving it there?

Reply:
Sure, I can do that.